### PR TITLE
Cosmoz tabbed template behavior

### LINKED ISSUE #44: Add template support for cosmoz-tab component
`cosmoz-tab` should support a template that will get stamped the first time the tab is activated.
For now `cosmoz-templatize` can be used until `1.x` is fully dropped.
Previous behavior of `cosmoz-tab` should remain unchanged.

##### How:

- [ ] On `attached` start observing newly added elements
- [ ] If a `template` is added that will be the one to use and we can templatize it
- [ ] On `tab-first-select` event stamp a child and add it the DOM.
- [ ] Setup all callback to forward data up a and down between the tab and the template instance (`forwardHostProp`, `notifyInstanceProp` and 1.x versions)
- [ ] Add a `prerender` Boolean property. If that is true when the template is added or when it becomes true and the template is not yet stamped proceed to stamp the template.

##### Considerations:
- [ ] Template stamping should be async (`requestIdleCallback` or  timeout +  `requestAnimationFrame`). It should not lock the event loop/thread. 
- [ ] On detach extra cleanup might be necessary to allow GC to collect the template instance.

### PR DESCRIPTION
**Pull request by programmer4web**
fixes #44

Test in cosmoz-frontend, example of use in one view: https://github.com/Neovici/cosmoz-frontend/pull/236

Implement tests for 'tabbed template behavior' data binding. Done.

### CODE REVIEW

File: cosmoz-tabbed-template-behavior.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			this._observer = null;
+			this._templateInstance = null;
+		},
+
+		_onNodesChanged() {
+			if (this._userTemplate) {
+				return;
+			}
+			const template = this.queryEffectiveChildren('template');
+			if (!template) {
```
Comment (megheaiulian) [inline cosmoz-tabbed-template-behavior.html]: This is not needed because `cosmoz-tab` should work with direct children and no template.

Comment (programmer4web) [reply cosmoz-tabbed-template-behavior.html]: no template warning removed in commit 569ec6d

**Comment by programmer4web** [comment]
@nomego This is ready for Review.

File: cosmoz-tabbed-template-behavior.html [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	/**
+	* @polymerBehavior Cosmoz.TabbedTemplateBehavior
+	*/
+	Cosmoz.TabbedTemplateBehaviorImpl = {
+		properties: {
+
+			/**
+			 * If it is set to true the template will be stamped
+			 */
+			prerender: {
```
Comment (nomego) [inline cosmoz-tabbed-template-behavior.html]: What's the difference between a prerendered template and a direct child?
Seems like the use cases would be the same?

**Pull request merged**